File: primerosProgramasPooo/Fraccion.java
package primerosProgramasPoo;

public class Fraccion {
	int numerador;
	int denominador;
	
	//constructor
	public Fraccion(int num, int denom) {
		numerador=num;
		denominador=denom;
	}
	/**metodo para obtener la fraccion.
	 * 
	 * @return String
	 * 
	 * @author Rafael Infante
	 * */
	
	public String obtenerFraccion() {
		return "La fraccion obtenida es: " + this.numerador + "/" + this.denominador;
	}
	/**metodo que modifica el numerador y denominador de la fraccion.
	 * 
	 * @param: int numerador
	 * @param: int denominador
	 * 
	 * @author Rafael Infante
	 * */
	
	public void modificarFraccion(int num, int den) {
		if(den==0) {
			System.out.println("No se puede dividir el denominador es igual a 0");
		}else {
			this.numerador=num;
			this.denominador=den;
		}
		
	}
	/**metodo que en el que obtienes el resultado de dividir numerador y 
	 * denominador de la fraccion.
	 * 
	 * @return int
	 * 
	 * @author Rafael Infante
	 * */
	
	public int resultadoFraccion() {
		return this.numerador/this.denominador;
	}
	/**metodo que multiplica la fraccion por un numero.
	 * 
	 * @param: int numero
	 * 
	 * @author Rafael Infante
	 * */
	
	public void multiplicaFraccionPorUnnumero(int num) {
		this.numerador *= num;
	}
	/**metodo que resta fracciones.
	 * 
	 * @param: int numerador
	 * @param: int denominador
	 * 
	 * @author Rafael Infante
	 * */
	
	public void restarFracciones(int num,int den) {
		//multiplicamos los denominadores
		int var3=this.denominador*den;
		//calculamos el resultado del primer numerador
		int var1=var3/this.denominador;
		var1*=this.numerador;
		//calculamos el resultado del segundo numerador
		int var2=var3/den;
		var2*=num;
		//restamos los numeradores
		this.numerador=var1-var2;
		this.denominador=var3;
	}
	/**metodo que suma fracciones.
	 * 
	 * @param: int numerador
	 * @param: int denominador
	 * 
	 * @author Rafael Infante
	 * */
	
	public void sumarFracciones(int num,int den) {
		//multiplicamos los denominadores
		int var3=this.denominador*den;
		//calculamos el resultado del primer numerador
		int var1=var3/this.denominador;
		var1*=this.numerador;
		//calculamos el resultado del segundo numerador
		int var2=var3/den;
		var2*=num;
		//sumamos los numeradores
		this.numerador=var1+var2;
		this.denominador=var3;
	}
	/**metodo que multiplica fracciones.
	 * 
	 * @param: int numerador
	 * @param: int denominador
	 * 
	 * @author Rafael Infante
	 * */
	
	public void multiplicaFracciones(int num, int den) {
		this.numerador *= num;
		this.denominador*=den;
	}
	
	public int hallarMcd() {
		int dividendo;
		int divisor;
		dividendo=this.numerador;
		divisor=this.denominador;
		while(dividendo%divisor!=0) {
			int resto = dividendo%divisor;
			dividendo=divisor;
			divisor=resto;
		}
		
		return divisor;
	}
	
	public void simplificarFraccion() {
		int mcd=hallarMcd();
		this.numerador/=mcd;
		this.denominador/=mcd;

	}
	
	
}

File: primerosProgramasPoo/Fraccion.py
'''
Crea una clase Fraccion (en Java yPython) de forma que podamos hacer las siguientes operaciones:

  Contruir un objeto Fraccion pasándole al constructor el numerador y el denominador.
  Obtener la fracción.
  Obtener y modificar numerador y denominador. No se puede dividir por cero.
  Obtener resultado de la fracción (número real).
  Multiplicar la fracción por un número.
  Multiplicar, sumar y restar fracciones.
  Simplificar la fracción.

@version:1.0
@author: Rafael Infante

'''

class Fraccion:
  
  '''
  constructor le pasamos como argumentos numerador y denominador
  
  @param: numerador
  @param: denominador
  '''
  def __init__(self,numerador,denominador):
    self.numerador=numerador
    self.denominador=denominador
    
  '''
  funcion para obtener la fraccion.
   
  @return: String
  '''
  def obtenerFraccion(self):
    return str(self.numerador) + "/" + str(self.denominador)
  
  '''
  funcion que modifica el numerador y denominador de la fraccion.
    
  @param: numerador
  @param: denominador
  '''
  def modificaFraccion(self,numerador,denominador):
    if denominador==0:
      print("No se puede dividir el denominador es igual a 0.")
    else:
      self.numerador=numerador
      self.denominador=denominador
  
  '''
  metodo que en el que obtienes el resultado de dividir numerador y 
  denominador de la fraccion.
   
  @return: un numero entero
  '''
  def resultadoFraccion(self):
    return self.numerador/self.denominador
  
  '''
  metodo que multiplica la fraccion por un numero.
  
  @param: un numero entero
  '''
  def multiplicaFraccionPorUnNumero(self,numero):
    self.numerador*=numero
 
  '''
  metodo que resta fracciones.
  
  @param: int numerador
  @param: int denominador
  '''
  def restarFracciones(self,numerador,denominador):
    if self.denominador == denominador:
      self.numerador-=numerador
    else:
      #multiplicamos los denominadores
      var3=self.denominador*denominador
      #calculamos el resultado del primer numerador
      var1=var3/self.denominador
      var1*=self.numerador
      #calculamos el resultado del segundo numerador
      var2=var3/denominador
      var2*=numerador
      #restamos los numeradores
      self.numerador=int(var1-var2)
      self.denominador=var3
    
  '''
  metodo que suma fracciones.
  
  @param: int numerador
  @param: int denominador
  '''
  def sumarFracciones(self,numerador,denominador):
    if self.denominador == denominador:
      self.numerador-=numerador
    else:
      #multiplicamos los denominadores
      var3=self.denominador*denominador
      #calculamos el resultado del primer numerador
      var1=var3/self.denominador
      var1*=self.numerador
      #calculamos el resultado del segundo numerador
      var2=var3/denominador
      var2*=numerador
      #restamos los numeradores
      self.numerador=int(var1+var2)
      self.denominador=var3
    
  '''
  metodo que multiplica fracciones.
  
  @param: int numerador
  @param: int denominador
  '''
  def multiplicaFracciones(self,numerador,denominador):
    self.numerador *= numerador
    self.denominador*=denominador
  
  '''
  halla el M.C.D de los denominadores de las fracciones
  '''
  def hallarMcd(self):
    dividendo=self.numerador
    divisor=self.denominador
    while dividendo%divisor!=0:
      resto = dividendo%divisor
      dividendo=divisor
      divisor=resto
    return divisor
  
  '''
  Funcion que simplifica fracciones al maximo
  '''
  def simplificarFraccion(self):
    mcd=self.hallarMcd()
    self.numerador=self.numerador//mcd
    self.denominador=self.denominador//mcd
  